Mesh_3/include/CGAL/Mesh_criteria_3.h

```diff
@@ -150,56 +125,181 @@ class Mesh_criteria_3_impl
+                              parameters::facet_distance=0.1,
+                              parameters::cell_radius_edge_ratio=2,
+                              parameters::cell_size=1.5);
+
+// Create a Mesh_criteria_3<Tr> object with size ignored (note that the order changed)
+Mesh_criteria_3<Tr> criteria (parameters::cell_radius_edge_ratio=2,
+                              parameters::facet_angle=30,
+                              parameters::facet_distance=0.1);
+
+\endcode
+
+\sa `MeshCriteria_3`
+\sa `MeshCriteriaWithFeatures_3`
+\sa `MeshCellCriteria_3`
+\sa `MeshEdgeCriteria_3`
+\sa `MeshFacetCriteria_3`
+\sa `MeshDomainField_3`
+\sa `CGAL::Mesh_cell_criteria_3<Tr>`
+\sa `CGAL::Mesh_edge_criteria_3<Tr>`
+\sa `CGAL::Mesh_facet_criteria_3<Tr>`
+\sa `CGAL::Mesh_facet_topology`
+
+*/
 template <typename Tr,
           typename EdgeCriteria = Mesh_edge_criteria_3<Tr>,
           typename FacetCriteria = Mesh_facet_criteria_3<Tr>,
           typename CellCriteria = Mesh_cell_criteria_3<Tr> >
 class Mesh_criteria_3
+#ifndef DOXYGEN_RUNNING
   : public internal::Mesh_criteria_3_impl< Tr,
                                            EdgeCriteria,
                                            FacetCriteria,
                                            CellCriteria >
+#endif
 {
   typedef internal::Mesh_criteria_3_impl< Tr,
                                           EdgeCriteria,
                                           FacetCriteria,
                                           CellCriteria>   Base;
 
 public:
+#ifdef DOXYGEN_RUNNING
+/// \name Types
+/// @{
+
+/*!
+The criteria for edges.
+*/
+typedef Mesh_edge_criteria_3<Tr> Edge_criteria;
+
+/*!
+The criteria for facets.
+*/
+typedef Mesh_facet_criteria_3<Tr> Facet_criteria;
+
+/*!
+The
+criteria for cells.
+*/
+typedef Mesh_cell_criteria_3<Tr> Cell_criteria;
+
+/// @}
+#else
   typedef typename Base::Edge_criteria    Edge_criteria;
   typedef typename Base::Facet_criteria   Facet_criteria;
   typedef typename Base::Cell_criteria    Cell_criteria;
+#endif
 
-  // Constructor
+  /// Construction from facet and cell criteria, the edge criteria are ignored in this case.
   Mesh_criteria_3(Facet_criteria facet_criteria,
                   Cell_criteria cell_criteria)
     : Base(facet_criteria,
            cell_criteria) {}
 
-  // Constructor
+  /// Constructor from edge, face and cell criteria.
   Mesh_criteria_3(Edge_criteria edge_criteria,
                   Facet_criteria facet_criteria,
                   Cell_criteria cell_criteria)
     : Base(edge_criteria,
            facet_criteria,
            cell_criteria) {}
+/*!
+ * \brief Construction from criteria parameters.
+ *
+ * Note that each size or distance parameter can be specified using two ways: either as a scalar field or
+ * as a numerical value when the field is uniform.
+ *
+ * If not specified, each parameter has a default value such that the corresponding criterion will be ignored.
```

Comment:
Actually, most parameters have 0 as default value, such that they are ignored if not specified. This is the case for all criteria *but* `facet_topology` that has `CGAL::FACET_VERTICES_ON_SURFACE` as default value. @lrineau Do you think the documentation clear enough about that?

Reply:
Yes, I think the documentation of the parameter `facet_topology` at https://cgal.github.io/6642/v0/Mesh_3/classCGAL_1_1Mesh__criteria__3.html is clear enough: the default is mentioned.

Maybe sentence you point out should have a "but `facet_topology`" somewhere.